openlibrary/solr/query_utils.py

```diff
@@ -279,3 +278,62 @@ def luqum_replace_field(query, replacer: Callable[[str], str]) -> str:
         if isinstance(sf, SearchField):
             sf.name = replacer(sf.name)
     return str(query)
+
+
+def luqum_field_traverse(item: Item, _parents: list[Item] | None = None):
+    """
+    Unlike the normal field traversal, this function will not go deeper once it hits a search field.
+    THis is necessary, in order to properly avoid fuzzying out undesired fields in luqum_make_fuzzy.
+
+    Nuch like its sister function, it does not make any guarantees about what will happen if you
+    modify the tree while traversing it 😅 But we do it anyways.
+
+    :param item: Node to traverse
+    :param _parents: Internal parameter for tracking parents
+    """
+    parents = _parents or []
+    yield item, parents
+    new_parents = [*parents, item]
+    if not isinstance(item, SearchField):
```

Comment:
Since this is the only difference between this and `luqum_traverse`, adding a new parameter to `luqum_traverse` , maybe `stop_at: Tuple[Class] = None` which we can set to SearchField might be a useful way to DRY these.

Reply:
Done, but changed to a Set from a tuple, so that the check for inclusion is quicker.